plugins/catalog-backend/src/service/DefaultLocationService.ts

```diff
@@ -98,6 +98,14 @@ export class DefaultLocationService implements LocationService {
       });
 
       if (processed.ok) {
+        const { metadata, kind } = processed.completedEntity;
+        if (
+          entities.some(
+            e => e.metadata.name === metadata.name && e.kind === kind,
+          )
+        ) {
+          throw new Error(`Duplicate nested entity: ${metadata.name}`);
```

Comment:
Can you check how this is handled on  the calling site. If this stops the whole location processing, I'm not certain if it's a good thing or not. Could be (I'm not sure, I don't know how it behaves) better to skip silently instead?